liquibase-cli/src/main/java/liquibase/integration/commandline/LiquibaseCommandLine.java

```diff
@@ -0,0 +1,789 @@
+import liquibase.exception.CommandLineParsingException;
+import liquibase.exception.CommandValidationException;
+import liquibase.hub.HubConfiguration;
+import liquibase.integration.IntegrationConfiguration;
+import liquibase.license.LicenseServiceFactory;
+import liquibase.logging.LogMessageFilter;
+import liquibase.logging.LogService;
+import liquibase.logging.core.JavaLogService;
+import liquibase.resource.CompositeResourceAccessor;
+import liquibase.resource.FileSystemResourceAccessor;
+import liquibase.ui.ConsoleUIService;
+import liquibase.util.LiquibaseUtil;
+import liquibase.util.StringUtil;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.*;
+import java.util.logging.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.ResourceBundle.getBundle;
+import static liquibase.util.SystemUtil.isWindows;
+
+
+public class LiquibaseCommandLine {
+
+    private final Map<String, String> legacyPositionalArguments;
+
+    /**
+     * Arguments that used to be global arguments but are now command-level
+     */
+    private final Set<String> legacyNoLongerGlobalArguments;
+
+    /**
+     * Arguments that used to be command arguments but are now global
+     */
+    private final Set<String> legacyNoLongerCommandArguments;
+    private Level configuredLogLevel;
+
+    private final CommandLine commandLine;
+    private FileHandler fileHandler;
+
+    private final ResourceBundle coreBundle = getBundle("liquibase/i18n/liquibase-core");
+
+    /**
+     * Pico's defaultFactory does a lot of reflection, checking for classes we know we don't have.
+     * That is slow on older JVMs and impact initial startup time, so do our own factory for performance reasons.
+     * It is easy to configure pico to it's default factory, when profiling check for `CommandLine$DefaultFactory` usage
+     */
+    private CommandLine.IFactory defaultFactory = new CommandLine.IFactory() {
+        @Override
+        public <K> K create(Class<K> cls) throws Exception {
+            return cls.newInstance();
+        }
+    };
+
+    public static void main(String[] args) {
+        //we don't ship jansi, so we know we can disable it without having to do the slow class checking
+        System.setProperty("org.fusesource.jansi.Ansi.disable", "true");
+        final LiquibaseCommandLine cli = new LiquibaseCommandLine();
+
+        int returnCode = cli.execute(args);
+
+        System.exit(returnCode);
+    }
+
+    private void cleanup() {
+        if (fileHandler != null) {
+            fileHandler.flush();
+            fileHandler.close();
+        }
+    }
+
+    public LiquibaseCommandLine() {
```

Comment:
Do the updateCount/Tag and changelogSyncToTag commands need to be in the legacyPositionalArguments?

Reply:
fixed